### ISSUE TITLE
Read the request type

### ISSUE THREAD

**Issue by djw8605** [scitokens/apache-scitokens#2]
The request type should be read in order to determine if "read", or "write" permissions are needed.  For example, a `PUT` or `POST` should require "write" permission.

https://github.com/scitokens/apache-scitokens/blob/0cf0aa9c3a5eece7b2ca4295926ca8fd91cb4b58/src/scitoken.c#L194-L196

**Comment by ag012** [comment]
read/write permission is now a setting that can be configured in the apache config file

**Comment by djw8605** [comment]
The request type is not something that is configured.  Instead, each request to a web server is a type, such as `GET`, `PUT`, `POST`.  Each of these implies either read or write.  The read or write should be from the request type, it's not something you can configure.

**Comment by ag012** [comment]
Got it. If that is the case, I am considering using a dictionary that has all methods of HTTP/WebDAV request and their corresponding resource. Would that be something that will work?

**Comment by djw8605** [comment]
Yup, that would work.

**Comment by ag012** [comment]
Thanks!
Pushed a fix.
I was a little worried about whether I found all the possible request types. Here is the list I used. If a request type is not found, the module will not grant any access.
"COPY:Write DELETE:Write GET:Read HEAD:Read LOCK:Write MKCOL:Write MOVE:Write OPTIONS:Read POST:Read PROPFIND:Write PROPPATCH:Write PUT:Write TRACE:Read UNLOCK:Write”;

I messed up the commit log a little bit and will fix that
